datar/apis/dplyr.py

```diff
@@ -2478,3 +2478,43 @@ def any_of(_data, x, vars=None) -> Any:
             in `_data` columns
     """
     raise _NotImplementedByCurrentBackendError("any_of", _data)
+
+
+@_register_verb(object)
+def pipe(_data: T, func: _Callable, *args, **kwargs) -> Any:
+    """Apply a function to the data
+
+    This function is similar to pandas.DataFrame.pipe() and allows you to
+    apply custom functions in a piping workflow. Works with any data type.
+
+    Args:
+        _data: The data object (can be any type)
+        func: Function to apply to the data. ``args`` and ``kwargs`` are
+            passed into ``func``.
+        *args: Positional arguments passed into ``func``
+        **kwargs: Keyword arguments passed into ``func``
+
+    Returns:
+        The return value of ``func``
+
+    Examples:
+        >>> import datar.all as dr
+        >>> # Works with lists
+        >>> [1, 2, 3] >> dr.pipe(lambda x: [i * 2 for i in x])
+        [2, 4, 6]
+
+        >>> # Works with dicts
+        >>> {'a': 1, 'b': 2} >> dr.pipe(lambda x: {k: v * 2 for k, v in x.items()})
+        {'a': 2, 'b': 4}
+
+        >>> # With additional arguments
+        >>> def add_value(data, value):
+        ...     return [x + value for x in data]
+        >>> [1, 2, 3] >> dr.pipe(add_value, 10)
+        [11, 12, 13]
+
+        >>> # Chain multiple operations
+        >>> [1, 2, 3] >> dr.pipe(lambda x: [i * 2 for i in x]) >> dr.pipe(sum)
+        12
+    """
+    return func(_data, *args, **kwargs)
```

Comment:
This should be under datar/misc.py.

It should be in any of the apis because it doesn't need to be implemented by the backends.